### PR TITLE
[Sprint] [FPM] Fix some issues in ASR

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
02f81053e8480c40d1291228ebb14a8b4d7e5c7b and b757048380bdc488d0448f5aac019a20bd37bee0

### CODE REVIEW

**Comment by Thirumalai-Shaktivel** [comment]
I came across this:
```fortran
program x
print *, max(1,2,3)
print *, max(1,2,3,4,5,6)
end program
```
Output:
```py
2
2
```
@czgdp1807 do you get the same output?

File: src/runtime/pure/lfortran_intrinsic_math2.f90 [hunk 1]
```diff
@@ -397,6 +397,11 @@ elemental integer(i64) function imax64(x, y) result(r)
 end if
 end function
 
+elemental integer function imax_3_args(x, y, z) result(r)
+integer, intent(in) :: x, y, z
+r = imax(imax(x, y), z)
+end function
```
Comment (Thirumalai-Shaktivel) [inline src/runtime/pure/lfortran_intrinsic_math2.f90:403]: Even after modifying the above code to the following, I get the same output.
```suggestion
elemental integer function imax_3_args(x, y, z) result(r)
integer, intent(in) :: x, y, z
if (x > y .and. x > z) then
    r = x
else if (y > z) then
    r = y
else
    r = z
end if
end function
```

**Comment by certik** [comment]
I think this change looks good to me.

File: src/runtime/pure/lfortran_intrinsic_math2.f90 (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply src/runtime/pure/lfortran_intrinsic_math2.f90]: I'm moving this to a separate PR.

**Comment by Thirumalai-Shaktivel** [comment]
Now, this PR is ready to go in.

File: src/runtime/pure/lfortran_intrinsic_math2.f90 (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply src/runtime/pure/lfortran_intrinsic_math2.f90]: https://github.com/lfortran/lfortran/pull/1134

**Review by certik — APPROVED** [review]
Thanks!

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the approval.

**Pull request merged**